src/preamble.js

```diff
@@ -796,6 +796,75 @@ function createExportWrapper(name, fixedasm) {
 }
 #endif
 
+#if ABORT_ON_WASM_EXCEPTIONS
+// When DISABLE_EXCEPTION_CATCHING != 1 `abortWrapperDepth` counts the recursion 
+// level of the wrapper function so that we only handle exceptions at the top level
+// letting the exception mechanics work uninterrupted at the inner level.
+// Additionally, `abortWrapperDepth` is also manually incremented in callMain so that 
+// we know to ignore exceptions from there since they're handled by callMain directly.
+var abortWrapperDepth = 0;
+
+// Instrument all the exported functions to:
+// - abort if an unhandled exception occurs
+// - throw an exception if someone tries to call them after the program has aborted
+// See settings.ABORT_ON_WASM_EXCEPTIONS for more info.
+function instrumentWasmExportsWithAbort(exports) {
+  var instExports = {};
+  for (var name in exports) {
+    (function(name) {
+      var original = exports[name];
+      
+      // Wrap all functions, copy the other symbols.
+      if (typeof original === 'function') {
+        instExports[name] = function() {
+          // Don't allow this function to be called if we're aborted!
+          if (ABORT == 1) { 
+#if ASSERTIONS
+            throw "program has already aborted!";
+#else
+            throw "DEAD";	
+#endif
+          }
+          
+#if DISABLE_EXCEPTION_CATCHING != 1
+          abortWrapperDepth += 1;
+#endif
+          try {
+            return original.apply(null, arguments);
+          }
+          catch (e) {
+            if(
```

Comment:
space after `if` please
```suggestion
            if (
```

Reply:
😱 I did it again!